### PR TITLE
[16912] Fast DDS introduction section

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
Depends on eProsima/ros2_documentation#15

### CODE REVIEW

File: docs/rst/introduction/introduction.rst [hunk 1]
```diff
@@ -21,15 +21,15 @@ Middleware
 The middleware layer is charged with the communication between components.
 After considering several communication protocols the `Object Management Group (OMG) <https://www.omg.org/>`_ `DDS (Data Distribution Service) <https://www.omg.org/spec/DDS/About-DDS/>`_ protocol was selected for being an open standard with a security extension, having a distributed dynamic discovery and being highly customizable for every kind of application.
 Several DDS vendors, both open source and licensed, are available within the *ROS 2* environment.
-However, *Vulcanexus* supports `eProsima's Fast DDS <https://fast-dds.docs.eprosima.com/en/latest/>`_ open source implementation.
+However, *Vulcanexus* supports `eProsima's Fast DDS <vulcanexus_fastdds>`_ open source implementation.
```
Comment (EduPonz) [inline docs/rst/introduction/introduction.rst:24]:
```suggestion
However, *Vulcanexus* supports :ref:`eProsima's Fast DDS <vulcanexus_fastdds>` open source implementation.
```

File: docs/rst/introduction/middleware.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
+.. _vulcanexus_middleware:
+
+Vulcanexus MiddleWare
```
Comment (EduPonz) [inline docs/rst/introduction/middleware.rst:3]:
```suggestion
Vulcanexus Middleware
```

File: docs/rst/introduction/middleware.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  * Allows to preallocate resources, to minimize dynamic resource allocation.
+  * Avoids the use of unbounded resources.
+  * Minimizes the need to copy data.
+
+* **Multi-platform:** The OS dependencies are treated as pluggable modules.
+  Users may easily implement platform modules using the *eProsima Fast DDS* library on their target platforms.
+  By default, the project can run over Linux, Windows and MacOS.
+
+* **Free and Open Source:** *eProsima Fast DDS*, the internal dependencies (such as *eProsima Fast CDR*) and the external ones (such as the *foonathan* library) are free and open source.
```
Comment (EduPonz) [inline docs/rst/introduction/middleware.rst:69]:
```suggestion
* **Free and Open Source:** Both *eProsima Fast DDS*, its internal dependencies (such as *eProsima Fast CDR*), and its external ones (such as the *foonathan* library), are free and open source.
```

File: docs/rst/introduction/middleware.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+Fast DDS
+--------
+
+*eProsima Fast DDS* is a C++ implementation of the `DDS (Data Distribution Service) Specification <https://www.omg.org/spec/DDS/About-DDS/>`__, a protocol defined by the `Object Management Group (OMG) <https://www.omg.org/>`__.
+The *eProsima Fast DDS* library provides both an Application Programming Interface (API) and a communication protocol that deploy a Data-Centric Publisher-Subscriber (DCPS) model, with the purpose of establishing efficient and reliable information distribution among Real-Time Systems.
+*eProsima Fast DDS* is predictable, scalable, flexible, and efficient in resource handling.
+For meeting these requirements, it makes use of typed interfaces and hinges on a many-to-many distributed network paradigm that neatly allows separation of the publisher and subscriber sides of the communication.
+
+Main Features
```
Comment (EduPonz) [inline docs/rst/introduction/middleware.rst:21]: I think it'd be neat if we added some links to the already existing Vulcanexus tutorials and docs in some of these bullet points.

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by JLBuenoLopez-eProsima** [comment]
Suggestions applied. Be aware that a new label has been included in ROS 2 documentation fork (eProsima/ros2_documentation#15) so CI is going to fail until that PR is merged. Then a last commit has to be included in this PR updating the submodule.

**Review by EduPonz — APPROVED** [review]

**Pull request merged**